deepchecks/core/display.py

```diff
@@ -244,23 +247,26 @@ def display_in_gui(result: DisplayableResult):
             'or use "result.save_as_html()" to save result'
         )
     else:
-        try:
-            app = QApplication(sys.argv)
-            web = QWebEngineView()
-            web.setWindowTitle('deepchecks')
-            web.setGeometry(0, 0, 1200, 1200)
-
-            html_out = io.StringIO()
-            result.save_as_html(html_out)
-            web.setHtml(html_out.getvalue())
-            web.show()
-
-            sys.exit(app.exec_())
-        except BaseException:  # pylint: disable=broad-except
-            get_logger().error(
-                'Unable to show result, run in an interactive environment '
-                'or use "result.save_as_html()" to save result'
-            )
+        filename = t.cast(str, result.save_as_html('deepchecks-report.html'))
```

Comment:
`setHtml` has a limitation - [link](https://doc.qt.io/qtforpython-5/PySide2/QtWebEngineWidgets/QWebEngineView.html#PySide2.QtWebEngineWidgets.PySide2.QtWebEngineWidgets.QWebEngineView.setHtml)

>Content larger than 2 MB cannot be displayed, because converts the provided HTML to percent-encoding and places data : in front of it to create the URL that it navigates to. Thereby, the provided code becomes a URL that exceeds the 2 MB limit set by Chromium. If the content is too large, the [loadFinished()](https://doc.qt.io/qtforpython-5/PySide2/QtWebEngineWidgets/QWebEngineView.html#PySide2.QtWebEngineWidgets.PySide2.QtWebEngineWidgets.QWebEngineView.loadFinished)signal is triggered with success=false .

plotly source is near 3 MB, this means if you inline javascript into HTML (set `connected=False`) it will not work

Reply:
Got it, so please save the file using tempfile

Reply:
read `tempfile` docs, it also has a lot of limitations and works differently on different platforms
in some cases, it even does not create a file

> Under Unix, the directory entry for the file is either not created at all or is removed immediately after the file is created. Other platforms do not support this; your code should not rely on a temporary file created using this function having or not having a visible name in the file system.

Reply:
take a look at my code, it has `finally` block, so the file is deleted at the end

Reply:
The docs you showed are from `TemporaryFile` but there is also `NamedTemporaryFile`. The only disadvantage is on windows you must close the file in order to read it again. Sounds more correct to me but if you delete the file anyway I guess I don't mind a lot

Reply:
oh, I did not see that,
but, yea, adding it will not change much